ui/pages/confirmations/components/confirm/nav/nav.tsx

```diff
@@ -0,0 +1,155 @@
+import { ethErrors, serializeError } from 'eth-rpc-errors';
+import React, { useCallback, useMemo } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { useHistory } from 'react-router-dom';
+
+import { ApprovalType } from '@metamask/controller-utils';
+
+import {
+  Box,
+  Button,
+  ButtonIcon,
+  ButtonIconSize,
+  ButtonVariant,
+  IconName,
+  Text,
+} from '../../../../../components/component-library';
+import {
+  AlignItems,
+  BackgroundColor,
+  BorderRadius,
+  Display,
+  FlexDirection,
+  FontWeight,
+  IconColor,
+  JustifyContent,
+  TextColor,
+  TextVariant,
+} from '../../../../../helpers/constants/design-system';
+import {
+  CONFIRM_TRANSACTION_ROUTE,
+  SIGNATURE_REQUEST_PATH,
+} from '../../../../../helpers/constants/routes';
+import { useI18nContext } from '../../../../../hooks/useI18nContext';
+import {
+  currentConfirmationSelector,
+  pendingConfirmationsSelector,
+} from '../../../../../selectors';
+import { rejectPendingApproval } from '../../../../../store/actions';
+
+const SignatureApprovalTypes = [
+  ApprovalType.EthSign,
+  ApprovalType.PersonalSign,
+  ApprovalType.EthSignTypedData,
+];
+
+const Nav = () => {
+  const history = useHistory();
+  const t = useI18nContext();
+  const currentConfirmation = useSelector(currentConfirmationSelector);
+  const pendingConfirmations = useSelector(pendingConfirmationsSelector);
+  const dispatch = useDispatch();
+
+  const currentConfirmationPosition = useMemo(() => {
+    if (pendingConfirmations?.length <= 0 || !currentConfirmation) {
+      return 0;
+    }
+    return pendingConfirmations.sort((a1, a2) => a1.time - a2.time).findIndex(
+      ({ id }) => id === currentConfirmation.id,
+    );
+  }, [currentConfirmation, pendingConfirmations]);
+
+  const onNavigateToTransaction = useCallback(
+    (pos: number) => {
+      const nextConfirmation =
+        pendingConfirmations[currentConfirmationPosition + pos];
```

Comment:
How is the next confirmation the one created after/before the current confirmation if pendingConfirmations is not ordered? :thinking: Is it a side effect of sorting the `pendingConfirmations` array when calling `currentConfirmationPosition`? If yes then we should definitely have a selector to get sorted pending confirmations.

Reply:
I agree, let me make these changes